### PR TITLE
Pre-multiplied alpha for the blend pass

### LINKED ISSUE #1354: Unexpected behavior when combining filters
I have observed that opacity filters do not combine as I would expect. For instance the two rectangles in the following wrench output are a different color:
```yaml
root:
  items:
    - type: stacking-context
      bounds: [10, 10, 500, 500]
      filters: [opacity(0.50), opacity(0.50)]
      items:
        - type: rect
          bounds: [10, 10, 100, 100]
          color: green
    - type: stacking-context
      bounds: [110, 10, 500, 500]
      filters: [opacity(0.25)]
      items:
        - type: rect
          bounds: [10, 10, 100, 100]
          color: green
```

screenshot

### LINKED ISSUE #1449: Opacity group compositing treats premultiplied RGBA as unpremultiplied RGBA
Testcase: [testcase](https://bugzilla.mozilla.org/attachment.cgi?id=8882833)
Screenshot:
screen shot 2017-07-01 at 2 35 25 pm
Firefox left, Servo right.

This box has a greyish fill in both servo and webrendest Gecko. It should have a yellow-whitish fill.

### PR DESCRIPTION
**Pull request by kvark**
Fixes #1449, fixes #1354
Running Servo tests is pending... Done.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/webrender/1493)

### CODE REVIEW

File: wrench/reftests/filters/opacity-combined.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# this tests that opacity combination respets pre-multiplied alpha
```
Comment (rlhunt) [inline wrench/reftests/filters/opacity-combined.yaml]: s/respets/respects

ditto in opacity-combined-ref.yaml

File: wrench/reftests/filters/opacity-combined-ref.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# this tests that opacity combination respets pre-multiplied alpha
+root:
+  items:
+    - type: stacking-context
+      bounds: [10, 10, 500, 500]
+      filters: [opacity(0.25)]
+      items:
+        - type: rect
+          bounds: [10, 10, 100, 100]
+          color: green
```
Comment (rlhunt) [inline wrench/reftests/filters/opacity-combined-ref.yaml]: Missing newline at end of file.

**Comment by rlhunt** [comment]
Thanks for adding tests! 👍

**Comment by kvark** [comment]
Hmm, maybe the extra space got in the way, trying again:
@bors-servo r=glennw,rhunt

**Comment by mstange** [comment]
Do the other functions in ps_blend.fs.glsl need to be adjusted to work with premultiplied alpha as well? For example the brightness filter may need to unpremultiply the input, apply the brightness factor to the rgb values, and then premultiply the output again.

**Comment by mstange** [comment]
I've filed #1507 and #1508 about my previous comment.

**Pull request merged**